src/klein/test/test_app.py

```diff
@@ -83,6 +108,38 @@ def foo(request):
         self.assertEqual(app.execute_endpoint("foo", DummyRequest(1)), "foo")
 
 
+    def test_mapByIdentity(self):
+        """
+        Routes are routed to the proper object regardless of its C{__hash__}
+        implementation.
+        """
+        a = DuplicateHasher("a")
+        b = DuplicateHasher("b")
+        self.assertEqual(a.myRouter.execute_endpoint("root", DummyRequest(1)),
+                         "a")
+        self.assertEqual(b.myRouter.execute_endpoint("root", DummyRequest(1)),
+                         "b")
+
+
+    def test_preserveIdentityWhenPossible(self):
+        """
+        Repeated accesses of the same L{Klein} attribute on the same instance
+        should result in an identically bound instance, when possible.
+        "Possible" is defined by a writable instance-level attribute named
+        C{__klein_bound_<the name of the Klein attribute on the class>__}.
```

Comment:
...so long as someone maintains a reference to the accessed object.  That means the following fails:

```python
def cacheSomething():
    a.myRouter.something = 1

def useSomething():
    print(a.myRouter.something + 1)

cacheSomething()
useSomething()
```

This is the current behavior, so it may not be worth mentioning.

It also includes accessing the same `Klein` object multiple times even if cannot be found as a class attribute, but that probably belongs in the test for `self._boundAs = 'unknown_' + str(id(self))`.